src/Host/Client/Impl/Host/RHost.cs

```diff
@@ -429,19 +429,32 @@ public sealed partial class RHost : IDisposable, IRExpressionEvaluator, IRBlobSe
                             case "!Plot":
                                 await _callbacks.Plot(
                                     new PlotMessage(
-                                        message.GetString(0, "xaml_file_path"),
-                                        message.GetInt32(1, "active_plot_index"),
-                                        message.GetInt32(2, "plot_count"),
+                                        message.GetString(0, "device_id").SafeParseGuid(),
```

Comment:
Do we actually want to ignore GUID parsing errors here? It looks like if it happens, it's a protocol contract violation (i.e. a "should never happen" case).